torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -376,6 +376,56 @@ bool IsBlockReturnTypeSame(Node* n) {
   return true;
 }
 
+void ComputeConstant(Node* n) {
+  switch (n->kind()) {
+    case ::c10::onnx::Shape: {
+      std::cout << "shape node: " << n->output()->debugName() << std::endl;      
+      std::cout << "shape input node: " << n->input()->debugName() << std::endl;
+      if (TensorShapeMap::HasShape(n->input()->debugName())) {
+        auto shape_size = TensorShapeMap::GetShape(n->input()->debugName()).concrete_sizes();
+        std::vector<int64_t> shape_value;
+        if (shape_size.has_value()) {
+          for (const auto& v: shape_size.value()) {
+            shape_value.emplace_back(static_cast<int64_t>(v));
+          }
+          ConstantTensorMap::SetValue(n->output()->debugName(), shape_value);
+        }
+      }
+      break;
+    }
+    case ::c10::onnx::Reshape: {
+      std::cout << "reshape node: " << n->output()->debugName() << std::endl;      
+      std::cout << "reshape input node: " << n->input(1)->debugName() << std::endl;
+      if (ConstantTensorMap::HasValue(n->input(1)->debugName())) {
+        auto shape_size = ConstantTensorMap::GetValue(n->input(1)->debugName());
+        auto vary_shape_size = c10::VaryingShape<int64_t>(shape_size);
+        TensorShapeMap::SetShape(n->output()->debugName(), vary_shape_size);
+        n->output()->setType(n->output()->type()->cast<TensorType>()->withSymbolicShapes(::c10::SymbolicShape(shape_size)));
+      }
+      break;
+    }
+    default: {
+      break;
+    }
+  }
+}
+
+void ProcessMap(Node* n) {
```

Comment:
Could we change it to a meaningful name, something like UpdateConstantOutputMapIfNecessary(), to avoid a general name which may confuse us in couple of months?